masonry/src/widgets/slider.rs

```diff
@@ -0,0 +1,854 @@
+// Copyright 2023 the Xilem Authors and the Druid Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use accesskit::{Node, Role};
+use cursor_icon::CursorIcon;
+use smallvec::SmallVec;
+use tracing::{trace_span, Span};
+use vello::{
+    kurbo::{Affine, Point, Rect, RoundedRect, RoundedRectRadii, Shape as _, Size},
+    peniko::Color,
+    Scene,
+};
+
+use crate::{
+    core::{
+        AccessCtx, AccessEvent, Action, BoxConstraints, EventCtx, LayoutCtx, PaintCtx,
+        PointerButton, PointerEvent, QueryCtx, RegisterCtx, TextEvent, Update, UpdateCtx, Widget,
+        WidgetId, WidgetMut,
+    },
+    theme,
+};
+
+use super::Axis;
+
+/// A slider widget for selecting a value within a range.
+pub struct Slider {
+    axis: Axis,
+    value: f64,
+    min: f64,
+    max: f64,
+    step: f64,
+    color: Color,
+    track_color: Color,
+    editing: bool,
+    is_dragging: bool,
+    grab_anchor: Option<f64>,
+    thumb_radii: RoundedRectRadii,
+    track_radii: RoundedRectRadii,
+    is_hovered: bool,
+    hover_glow_color: Color,
+    hover_glow_blur_radius: f64,
+    hover_glow_spread_radius: f64,
+    track_rect: Option<RoundedRect>,
+}
+
+// Constants
+impl Slider {
+    const DEFAULT_WIDTH: f64 = 200.0;
+    const DEFAULT_HEIGHT: f64 = 40.0;
+    const BASE_TRACK_THICKNESS: f64 = 4.0;
+    const TRACK_PADDING: f64 = 5.0;
+    const THUMB_WIDTH: f64 = 12.0;
+    const THUMB_HEIGHT: f64 = 20.0;
+}
+
+// Constructor and methods
+impl Slider {
+    /// Creates a new slider with the specified range and initial value.
+    pub fn new(axis: Axis, min: f64, max: f64, value: f64) -> Self {
+        debug_assert!(
+            max >= min,
+            "Slider max value must be greater than or equal to min value"
+        );
+        let (min, max) = (min.min(max), min.max(max));
+        Self {
+            axis,
+            value: value.clamp(min, max),
+            min,
+            max,
+            step: 1.0,
+            color: theme::PRIMARY_LIGHT,
+            track_color: theme::PRIMARY_DARK,
+            editing: false,
+            is_dragging: false,
+            grab_anchor: None,
+            thumb_radii: RoundedRectRadii::from_single_radius(5.0),
+            track_radii: RoundedRectRadii::from_single_radius(2.0),
+            is_hovered: false,
+            hover_glow_color: Color::from_rgba8(255, 255, 255, 50),
+            hover_glow_blur_radius: 5.0,
+            hover_glow_spread_radius: 2.0,
+            track_rect: None,
+        }
+    }
+
+    /// Sets the slider's color.
+    pub fn with_color(mut self, color: impl Into<Color>) -> Self {
```

Comment:
This should probably be with_thumb_color and rename the struct member as well.

also, should this and the other colors be a brush instead? That would permit gradients…